.github/workflows/release-binaries.yml

```diff
@@ -0,0 +1,188 @@
+        if: matrix.os == 'macOS-latest'
+        uses: actions-rs/cargo@v1.0.3
+        with:
+          command: build
+          args: --release --locked --target ${{ matrix.target }}
+
+      - name: Post Build | Prepare artifacts [Windows]
+        if: matrix.os == 'windows-latest'
+        run: |
+          cd target/${{ matrix.target }}/release
+          strip prometheus_wireguard_exporter.exe
+          mv prometheus_wireguard_exporter.exe ../../../${{ matrix.name }}
+          cd -
+
+      - name: Post Build | Prepare artifacts [-nix]
+        if: matrix.os != 'windows-latest'
+        run: |
+          cd target/${{ matrix.target }}/release
+          strip prometheus_wireguard_exporter || true
+          mv prometheus_wireguard_exporter ../../../${{ matrix.name }}
+          cd -
+
+      - name: Display structure of downloaded files
+        run: ls -R
+
+      - name: Deploy | Upload artifacts
+        uses: actions/upload-artifact@v2
+        with:
+          name: ${{ matrix.name }}
+          path: ${{ matrix.name }}
+
+  # Create GitHub release with Rust build targets and release notes
+  github_release:
+    name: Create GitHub Release
+    needs: github_build
+    runs-on: ubuntu-latest
+    steps:
+      - name: Setup | Checkout
+        uses: actions/checkout@v2.3.4
+        with:
+          fetch-depth: 0
+
+      - name: Setup | Go
+        uses: actions/setup-go@v2
+        with:
+          go-version: "^1.15.7"
+
+      - name: Setup | Download Artifacts
+        uses: actions/download-artifact@v2
+
+      - name: Display structure of downloaded files
+        run: ls -R
+
+      - name: Setup | Create Checksum files
+        run: |
+          for DIR in prometheus_wireguard_exporter-*
+          do
+            pushd "$DIR" || continue
+            FILE="$(echo prometheus_wireguard_exporter-*)"
+            sha256sum "$FILE" > "$FILE.sha256"
+            popd || exit
+          done
+
+      # - name: Setup | Release notes
+      #   run: |
+      #     GO111MODULE=on go get github.com/git-chglog/git-chglog/cmd/git-chglog@0.14.2
+      #     git-chglog -c .github/chglog/release.yml $(git describe --tags) > RELEASE.md
+
+      - name: Build | Publish binaries
+        uses: alexellis/upload-assets@0.3.0
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          asset_paths: '["prometheus_wireguard_exporter-*/prometheus_wireguard_exporter-*"]'
+
+  # Publish prometheus_wireguard_exporter to Crates.io
+  # cargo_publish:
+  #   name: Publish Cargo Package
+  #   runs-on: ubuntu-latest
+  #   needs: github_release
+  #   steps:
+  #     - name: Setup | Checkout
+  #       uses: actions/checkout@v2.3.4
+
+  #     - name: Setup | Rust
+  #       uses: actions-rs/toolchain@v1.0.7
+  #       with:
+  #         toolchain: stable
+  #         profile: minimal
+  #         override: true
+
+  #     - name: Build | Publish
+  #       run: cargo publish --token ${{ secrets.CRATES_IO_TOKEN }}
```

Comment:
should we remove that commented block? I don't think this repo is meant to be used as a crate right?

@MindFlavor what do you think?

Reply:
If possibile, it could be useful to automatically publish the crate to crates.io when publishing a release (I actually do it manually, *if* I remember to do it 😅).
There is no harm on doing it and allows people to install/upgrade the tool by issuing `cargo install prometheus_wireguard_exporter` if they so wish. 

It's nice to have but let's not sweat on it too much. The manual workflow is good enough 👍.

Reply:
Cool let's leave the commented block then, we can address it later on.